sharding/validatorDistributor.go

```diff
@@ -0,0 +1,39 @@
+package sharding
+
+// IntraShardValidatorDistributor - distributes validators from source to destination inside the same shard
+type IntraShardValidatorDistributor struct{}
+
+func (vd *IntraShardValidatorDistributor) DistributeValidators(
+	destination map[uint32][]Validator,
+	source map[uint32][]Validator,
+	_ []byte,
+) error {
+	moveNodesToMap(destination, source)
+	return nil
+}
+
+// IsInterfaceNil - verifies if the interface is nil
+func (vd *IntraShardValidatorDistributor) IsInterfaceNil() bool {
+	return vd == nil
+}
+
+// CrossShardValidatorDistributor - distributes validators from source to destination cross shards
+type CrossShardValidatorDistributor struct{}
+
+func (vd *CrossShardValidatorDistributor) DistributeValidators(
+	destination map[uint32][]Validator,
+	source map[uint32][]Validator,
+	rand []byte,
+) error {
+	allValidators := make([]Validator, 0)
+	for _, validatorList := range source {
+		allValidators = append(allValidators, validatorList...)
```

Comment:
you could use the moveNodesToMap function if you do not care about source. Or call distributeValidators with source instead of allValidators.

Reply:
moveNodesToMap works in the same shard. I wanted to create the "pool" and then distribute the validators.